fhir-server/src/main/java/au/csiro/pathling/fhirpath/function/translate/TranslateFunction.java

```diff
@@ -0,0 +1,213 @@
+ * translated using provided concept map URL.
+ * <p>
+ * Sinature:
+ * <pre>
+ * collection<Coding|CodeableConcept> -> translate(conceptMapUrl: string, reverse = false,
+ * equivalence = 'equivalent') : collection<Coding>
+ * </pre>
+ * <p>
+ * Uses: <a href="https://www.hl7.org/fhir/operation-conceptmap-translate.html">Translate
+ * Operation</a>
+ *
+ * @author Piotr Szul
+ * @see <a href="https://pathling.csiro.au/docs/fhirpath/functions.html#translate">translate</a>
+ */
+public class TranslateFunction implements NamedFunction {
+
+  private static final String NAME = "translate";
+
+  private static final Boolean DEFAULT_REVERSE = false;
+
+  private static final String DEFAULT_EQUIVALENCE = "equivalent";
+
+
+  /**
+   * Helper class for dealing with optional arguments.
+   */
+  private static class Arguments {
+
+    @Nonnull
+    private final List<FhirPath> arguments;
+
+    private Arguments(@Nonnull List<FhirPath> arguments) {
+      this.arguments = arguments;
+    }
+
+    /**
+     * Gets the value of an optional literal argument or the default value it the argument is
+     * missing.
+     *
+     * @param index the 0-based index of the argument.
+     * @param defaultValue the default value.
+     * @param <T> the Java type of the argument value.
+     * @return the java value of the requested argument.
+     */
+    @SuppressWarnings("unchecked")
+    @Nonnull
+    public <T> T getValueOr(int index, @Nonnull final T defaultValue) {
+      return (index < arguments.size())
+             ? getValue(index, (Class<T>) defaultValue.getClass())
+             : defaultValue;
+    }
+
+    /**
+     * Gets the value of the required literal argument.
+     *
+     * @param index the 0-based index of the argument.
+     * @param valueClass the expected Java  class of the argument value.
+     * @param <T> the Java type of the argument value.
+     * @return the java value of the requested argument.
+     */
+    @Nonnull
+    public <T> T getValue(int index, @Nonnull final Class<T> valueClass) {
+      return Objects
+          .requireNonNull(valueClass.cast(((LiteralPath) arguments.get(index)).getJavaValue()));
+    }
+
+    /**
+     * Construct {@link Arguments} for given {@link NamedFunctionInput}
+     *
+     * @param input the function input.
+     * @return the {@link Arguments} for the input.
+     */
+    @Nonnull
+    public static Arguments of(@Nonnull final NamedFunctionInput input) {
+      return new Arguments(input.getArguments());
+    }
+  }
+
+  @Nonnull
+  @Override
+  public FhirPath invoke(@Nonnull final NamedFunctionInput input) {
+    validateInput(input);
+
+    final ElementPath inputPath = (ElementPath) input.getInput();
+
+    final ParserContext inputContext = input.getContext();
+    final Column idColumn = inputPath.getIdColumn();
+    final Column conceptColumn = inputPath.getValueColumn();
+
+    final boolean isCodeableConcept = isCodeableConcept(inputPath);
+
+    CheckReturnValue functions;
```

Comment:
@piotrszul I'm not really familiar with this syntax - what is the purpose of this statement?

Reply:
I have no idea. I am guessing IntliJ must have inserted it.